Build on Windows with meson

mdpiper opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Build v0.3 on Windows.

mdpiper committed:
Enable Windows builds and bump build number

mdpiper committed:
Add Windows batch file

mdpiper commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. Your recipe may not receive automatic updates and/or may not be compatible with conda-forge's infrastructure. Please check the logs for more information and ensure your recipe can be parsed.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12207088182](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12207088182). Examine the logs at this URL for more detail.</sub>

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.44.9, and conda-forge-pinning 2024.12.05.17.22.05

mdpiper committed:
Remove unknown keyword argument

Because of bmi-fortran=1.2.1, I'm using an old version of flang,
which may be the problem.

mdpiper committed:
Set name of archiver for Windows

mdpiper committed:
Always require pkg-config

mdpiper committed:
Also include Python in build requirements
